File: Dec2020/30Dec/LicenseKeyFormatting.java
/*
Problem Name: 482. License Key Formatting
Problem Link: https://leetcode.com/problems/license-key-formatting/submissions/
*/
class Solution {
    public String licenseKeyFormatting(String S, int K) {
        S = S.toUpperCase();
        String ans = "";
        for(int i=S.length()-1,cnt=0;i>=0;i--){
            if(S.charAt(i) != '-'){
                ans = S.charAt(i) + ans;
                cnt++;
                if(cnt%K == 0){
                    ans = '-' + ans;
                    cnt = 0;
                }
            }
        }
        if(ans.length() == 0)
            return ans;
        if(ans.charAt(0) == '-')
            return ans.substring(1);
        return ans;
    }
}

File: Dec2020/30Dec/LicenseKeyFormatting.cpp
/*
Problem name: 482. License Key Formatting
Problem link: https://leetcode.com/problems/license-key-formatting/
*/

class Solution {
public:
    string licenseKeyFormatting(string S, int K) {
        string result="";
        int mark=0;
        for(int i=S.length()-1;i>=0;--i){
            if(S[i]!='-'){
                S[i]=toupper(S[i]);
                result += S[i];
                mark++;
                if(mark%K == 0){
                    result += '-';
                    mark=0;
                }
            }
            else
                continue;
        }
        reverse(result.begin(), result.end());
        if(result[0]=='-')
            return result.substr(1);
        return result;
    }
};
